io/writers_test.ts

```diff
@@ -22,3 +22,17 @@ Deno.test("ioStringWriterSync", function (): void {
   w.writeSync(encoder.encode("\nland"));
   assertEquals(w.toString(), "deno\nland");
 });
+
+Deno.test("ioStringWriterIsolationTest", async function () {
+  const encoder = new TextEncoder();
+  const src =  "ABC";
+  const srcChunks = src.split("").map(c => encoder.encode(c))
+
+  const w = new StringWriter();
+  for (const c of srcChunks) {
+    const written = await w.write(c);
+    assertEquals(written, 1)
+  }
```

Comment:
These chunks all been written... so manipulating their values shouldn't change the value of w.toString()